src/twister2/builder/build_manager.py

```diff
@@ -110,6 +122,9 @@ def build(self) -> None:
         if status == BuildStatus.DONE:
             logger.info('Already build in %s', self.build_config.build_dir)
             return
+        if status == BuildStatus.SKIPPED:
```

Comment:
I am not sure if we can assume that all skips are related to Kconfig and DTS filtering, now I always raise `TwisterKconfigFiltrationException` for skip, but maybe we should keep more information in build reports.